### PR TITLE
Created variants for app-input

### LINKED ISSUE #345: Add variants for input
image

### PR DESCRIPTION
**Pull request by D1nviz**
- Added variants
- Small enhancements

https://github.com/user-attachments/assets/e88cb8f7-4422-43bd-9b8d-bf2df2ca62ba

### CODE REVIEW

File: code/src/components/app-input/AppInput.scss [hunk 1]
```diff
@@ -1,12 +1,33 @@
...
-    }
-    .MuiFormLabel-root {
-      padding-top: 3px;
+$spa-input-colors: (
+  "contained": var(--spa-input-contained-color, $purple),
+  "success": var(--spa-input-success-color, $green),
+  "danger": var(--spa-input-danger-color, $red),
+  "light": var(--spa-input-light-color, $white),
+  "dark": var(--spa-input-dark-color, $black)
+);
```
Comment (dmtrth25) [inline code/src/components/app-input/AppInput.scss:9]: maybe create these variables like general for whole project ? 
These items also used for `AppSelect` maybe create some `common` and reuse it ? 
What do you think ?

File: code/src/components/app-input/AppInput.scss [hunk 2]
```diff
@@ -1,12 +1,33 @@
...
+
+.spa-input {
+  .MuiInputBase-input {
+    padding: spacing(3) spacing(4);
+  }
+
+  .MuiFormLabel-root {
+    padding-top: 3px;
```
Comment (dmtrth25) [inline code/src/components/app-input/AppInput.scss]: I suggest to add `spacing(1)` like with one approach everywhere. 1 px it's not so required I guess

File: code/src/components/app-input/AppInput.scss [hunk 3]
```diff
@@ -1,12 +1,33 @@
...
+  }
+
+  @each $variant, $color in $spa-input-colors {
+    &--#{$variant} {
+      .MuiOutlinedInput-root {
+        &.Mui-focused .MuiOutlinedInput-notchedOutline {
+          border-color: $color !important;
```
Comment (mhevyk) [inline code/src/components/app-input/AppInput.scss]: Add comment that is for overriding mui styles

File: code/src/components/app-input/AppInput.scss [hunk 4]
```diff
@@ -1,12 +1,33 @@
...
+        }
+      }
+
+      .MuiInputLabel-root.Mui-focused {
+        color: $color !important;
```
Comment (mhevyk) [inline code/src/components/app-input/AppInput.scss]: Same

File: code/src/components/app-input/AppInput.scss (hunk 3 above)
Comment (D1nviz) [reply code/src/components/app-input/AppInput.scss]: added

File: code/src/components/app-input/AppInput.scss (hunk 4 above)
Comment (D1nviz) [reply code/src/components/app-input/AppInput.scss]: added

File: code/src/components/app-input/AppInput.scss (hunk 2 above)
Comment (D1nviz) [reply code/src/components/app-input/AppInput.scss]: Refactored

File: code/src/components/app-input/AppInput.scss (hunk 1 above)
Comment (D1nviz) [reply code/src/components/app-input/AppInput.scss:9]: We already using global variables. Also, as you can see we are using different names for that. For example, here are `--spa-input-color`, in the app-select `--spa-select-color`.

**Review by mhevyk — APPROVED** [review]

**Review by dmtrth25 — APPROVED** [review]

**Review by heshvab — APPROVED** [review]

**Pull request merged**